pydicom/valuerep.py

```diff
@@ -77,14 +77,39 @@ def _range_regex(regex: str) -> str:
                    for (vr, regex) in VR_REGEXES.items()}
 
 
-def validate_vr_length(vr: str, value: Union[str, bytes]) -> Tuple[bool, str]:
+def validate_type(vr: str, value: Any,
+                  types: Union[Type, Tuple[Type, Type]]) -> Tuple[bool, str]:
+    """Checks for valid types for a given VR.
+
+    Parameters
+    ----------
+    vr : str
+        The value representation to validate against.
+    value : Any
+        The value to validate.
+    types: Type or Tuple[Type]
+        The type or tuple of types supported for the given VR.
+
+    Returns
+    -------
+        A tuple of a boolean validation result and the error message.
+    """
+    if value is not None and not isinstance(value, types):
+        return False, (
+            f"A value of type '{type(value).__name__}' cannot be "
+            f"assigned to a tag with VR {vr}."
```

Comment:
Maybe change it to include what types are allowed, along the lines of what Python does "X must be (required type), not (actual type)"